TriggerService/docker_context/triggerInputSchema.py

```diff
@@ -0,0 +1,60 @@
+from jsonschema import validate, ValidationError
+
+
+def validateRequest(data, schema):
+    if not data:
+        return False, 'Data is not Content-Type: application/json'
+
+    try:
+        validate(instance=data, schema=schema)
+        return True, ''
+    except ValidationError as err:
+        return False, str(err)
+
+
+def errorResult(err, data):
+    return {
+        'errorMessage': err,
+        'content': data
+    }
+
+
+one_to_three_letter_string = {
+    'type': 'string',
+    'minLength': 1,
+    'maxLength': 3
+}
+
+non_negative_number = {
+    'type': 'number',
+    'inclusiveMinimum': 0
+}
+
+base_transaction_schema = {
+    'type': 'object',
+    'properties': {
+        'transaction_num': {'type': 'integer'},
+        'user_id': {'type': 'string'},
+        'stock_symbol': one_to_three_letter_string,
+        'amount': non_negative_number
+    },
+    'required': ['transaction_num', 'user_id', 'stock_symbol', 'amount']
+}
+
+
+base_user_schema = {
+    'type': 'object',
+    'properties': {
+        'transaction_num': {'type': 'integer'},
+        'user_id': {'type': 'string'}
+    },
+    'required': ['transaction_num', 'user_id']
+}
+
+set_buy_amount_schema = base_transaction_schema
+set_buy_trigger_schema = base_transaction_schema
+cancel_buy_trigger_schema = base_user_schema

```

Comment:
Doesn't the buy trigger cancellation command also require a stock id in addition to a user id? The base user schema checks for transaction number and user id but not the stock id. Same with the sell trigger cancellation command. Please clarify if I am mistaken.

Reply:
that is correct thanks!